### ISSUE TITLE
Receiving 'NoClassDefFoundError: org/apache/commons/lang/StringUtils' 

### ISSUE THREAD

**Issue by google-code-exporter** [magro/memcached-session-manager#211]
_From [eravene...@gmail.com](https://code.google.com/u/106765032624795408949/) on July 15, 2013 20:10:00_

<b>What steps will reproduce the problem?</b>
&#13;
1. Drop jars in CATALINA_HOME/lib&#13;
2. Add &lt;Manager className="de.javakaffee.web.msm.MemcachedBackupSessionManager"&#13;
	memcachedNodes="memcachedserver:11211" &#13;
	requestUriIgnorePattern=".*\.(ico|png|gif|jpg|css|js)$" /&gt; to Context.xml&#13;
3. Start Tomcat&#13;
&#13;
&#13;
<b>What is the expected output? What do you see instead?</b>
&#13;
I expect Tomcat to start, memcached sessions to work.  Instead, I receive the following exception:&#13;
&#13;
java.lang.NoClassDefFoundError: org/apache/commons/lang/StringUtils&#13;
        at net.logstash.log4j.JSONEventLayout.format(JSONEventLayout.java:72)&#13;
        at org.apache.log4j.WriterAppender.subAppend(WriterAppender.java:310)&#13;
        at org.apache.log4j.DailyRollingFileAppender.subAppend(DailyRollingFileAppender.java:36&#13;
        at org.apache.log4j.WriterAppender.append(WriterAppender.java:162)&#13;
        at org.apache.log4j.AppenderSkeleton.doAppend(AppenderSkeleton.java:251)&#13;
        at org.apache.log4j.helpers.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttach&#13;
        at org.apache.log4j.Category.callAppenders(Category.java:206)&#13;
        at org.apache.log4j.Category.forcedLog(Category.java:391)&#13;
        at org.apache.log4j.Category.log(Category.java:856)&#13;
        at org.apache.juli.logging.impl.Log4JLogger.error(Log4JLogger.java:257)&#13;
        at org.apache.catalina.startup.Catalina.stopServer(Catalina.java:507)&#13;
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)&#13;
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)&#13;
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25&#13;
        at java.lang.reflect.Method.invoke(Method.java:597)&#13;
        at org.apache.catalina.startup.Bootstrap.stopServer(Bootstrap.java:371)&#13;
        at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:458)&#13;
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang.StringUtils&#13;
        at java.net.URLClassLoader$1.run(URLClassLoader.java:202)&#13;
        at java.security.AccessController.doPrivileged(Native Method)&#13;
        at java.net.URLClassLoader.findClass(URLClassLoader.java:190)&#13;
        at java.lang.ClassLoader.loadClass(ClassLoader.java:306)&#13;
        at java.lang.ClassLoader.loadClass(ClassLoader.java:247)&#13;
        ... 17 more&#13;
&#13;
&#13;
&#13;
<b>What version of the product are you using? On what operating system?</b>
&#13;
spymemcached-2.9.1.jar&#13;
memcached-session-manager-tc7-1.6.5.jar&#13;
memcached-session-manager-1.6.5.jar&#13;
&#13;
CentOS 6.3, Java 6 u45, Tomcat 7.0.39&#13;
&#13;
<b>Please provide any additional information below.</b>
&#13;
Removing my custom log4j layout jars and reverting to standard Tomcat 'out of the box' logging results in other exceptions that ultimately lead to the same problem with no server start.

_Original issue: http://code.google.com/p/memcached-session-manager/issues/detail?id=170_

**Comment by google-code-exporter** [comment]
_From [eravene...@gmail.com](https://code.google.com/u/106765032624795408949/) on July 15, 2013 11:34:10_

I downloaded the latest commons-lang jar from Apache, commons-lang3-3.1-bin.tar.gz, and placed the jar in my tomcat/lib folder which did not fix the issue.

Downloaded the older version, commons-lang-2.6-bin.tar.gz, and the issue has been resolved.  Obvious answer to an obvious problem.  I guess I shouldn't have assumed that commons lang was present (though why this exception occurred only after adding the session management when it apparently had nothing to do with it, who knows?).
